### PR TITLE
Executors documentation refactor

### PR DESCRIPTION
**Pull request by loujar**
- New doc page for executor configuration table (`/admin/executors/executors_config`)
  - previously, executor config documentation was nested under the `deploy_executors_binary` docs page, despite applying for all deployment methods
- Various edits to improve readability and flow of executor documentation

related PRs:
- https://github.com/sourcegraph/terraform-aws-executors/pull/93
- https://github.com/sourcegraph/terraform-google-executors/pull/107

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -92,7 +92,7 @@ For example the cards layout appears by using the `<Callout>` component that can
 You can use this component within your content as follows:
 
 ```js
-<Callout type="note">Cody is currently in Beta stage for all users.</Callout>
+<Callout type="note">Cody is currently in Beta for all users.</Callout>
```
Comment (MaedahBatool) [inline README.md:95]:
```suggestion

```
We should remove this as Cody is not in Beta anymore.

File: docs/admin/executors/deploy_executors.mdx [hunk 2]
```diff
@@ -1,20 +1,18 @@
...
+- [Terraform on AWS or GCP](/admin/executors/deploy_executors_terraform)
+- (Beta) [Native Kubernetes](/admin/executors/deploy_executors_kubernetes)
+- (Beta) [Docker-in-Docker on Kubernetes](/admin/executors/deploy_executors_dind)
+- (Beta) [Docker-Compose](/admin/executors/deploy_executors_docker)
 
-> NOTE: Executors are available with no additional setup required on Sourcegraph Cloud.
+See [deciding which executor deployment method to use ](../executors#deciding-which-executor-deployment-method-to-use) for more information on these different deployment options.
 
-## Requirements
+<Callout type="note">NOTE: Executors are available with no additional setup required on Sourcegraph Cloud.</Callout>
```
Comment (MaedahBatool) [inline docs/admin/executors/deploy_executors.mdx:13]:
```suggestion
<Callout type="note">Executors are available with no additional setup required on Sourcegraph Cloud.</Callout>
```

**Review by MaedahBatool — CHANGES_REQUESTED** [review]
Found small fixes. Overall looks good. Thank you for contributing. :)

**Comment by MaedahBatool** [comment]
Hey @loujar thanks a ton for opening this PR. I added some very small suggestions. Please merge the PR once you're ready with the approvals. :)

File: README.md (hunk 1 above)
Comment (loujar) [reply README.md:95]: Ah true, but I think this is just an example in the README for how to format `callout` components. This change is just to make the javascript example match the image above

Comment (MaedahBatool) [reply README.md:95]: Oh, my bad, I just missed this context. 🙈

**Review by MaedahBatool — APPROVED** [review]
Hey @loujar approving it LGTM 👍 Please merge once you are ready.

**Pull request merged**